### PR TITLE
Switch to CMake and flatten relative include paths.

### PR DESCRIPTION
**Pull request by zrax**
This should make reviving and actually working on imaginaryMyst easier.
TODO: Untangle circular dependencies...

### CODE REVIEW

**Comment by Deledrius** [comment]
No circular dependency issues in my conversion: 8772ca9fa07a183808c5308b54f6b07b0d2befca
(https://github.com/H-uru/imaginaryMyst/compare/cmake)

**Comment by zrax** [comment]
Sorry, I didn't realize you were working on the same thing...  :(

Questions though:  Why did you remove libsquish?  It's not provided by any Linux distros I'm aware of, and we still keep it in-source for libhsplasma...  Also, your version still doesn't fix the circular dependencies (look at the header files for those modules), it just doesn't mark them explicitly.  I had it without dependencies at first, but ran into link errors on Linux until I added them, which is when I noticed the circular dependency.

**Comment by Deledrius** [comment]
MSVC strikes again!  It seems to handle the circular dependencies without complaint. >_<  Silly me for assuming they were gone.

I removed libSquish since it appeared you hadn't made any repo-specific changes requiring it to be in-tree.  I also have a version that keeps it, if desired.  I'm not working in Linux right now, so I wasn't aware of the (lack of) ease of installation there; I simply built my own and installed it system-wide.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    imKeyList.cpp
+    imLog.cpp
+    imString.cpp
+    Main.cpp
+)
+
+source_group("Header Files" FILES ${imCommon_HEADERS})
+source_group("Source Files" FILES ${imCommon_SOURCES})
+
+add_executable(imaginaryMyst ${imCommon_HEADERS} ${imCommon_SOURCES})
```
Comment (Deledrius) [inline src/CMakeLists.txt]: We need `WIN32` here to link properly on Windows.

**Review by Deledrius — CHANGES_REQUESTED** [review]

**Review by Deledrius — APPROVED** [review]

**Pull request merged**